internal/backends/qt/qt_widgets/button.rs

```diff
@@ -190,7 +202,16 @@ impl NativeButton {
 }
 
 impl Item for NativeButton {
-    fn init(self: Pin<&Self>) {}
+    fn init(self: Pin<&Self>) {
+        let animation_tracker_property_ptr = Self::FIELD_OFFSETS.animation_tracker.apply_pin(self);
+        self.widget_ptr.set(cpp! { unsafe [animation_tracker_property_ptr as "void*"] -> QPushButtonPtr as "std::unique_ptr<QPushButton>" {
+            return make_unique_animated_widget<QPushButton>(animation_tracker_property_ptr);
+        }})
+    }
+
+    fn deinit(self: Pin<&Self>) {
+        self.widget_ptr.take();
```

Comment:
Why do we need this?  Doesn't the drop, or destructor, take care of this?

Reply:
For Rust that would be the case, yes. But not for C++ AFAICS.

Reply:
It should, since it is a std::unique_ptr in C++, its destructor should be called.

Reply:
Hmm, it isn't at the moment, but you have a point: I'll try to change the fields we generate for `QPushButtonPtr`, etc. in `cbindgen.rs`.

Reply:
Ah good point:

Idea: make it a `std::unique_ptr<QObject>`  (but still containing whatever widget it is)

Reply:
Hmm, that would help to avoid having to include `qpushbutton.h` and friends, but the core problem remains: Generated code depends on the availability of Qt headers in the include search path - if just for `qobject.h`.